File: src/MaksIT.CertsUI/Models/Identity/User/CreateUserRequest.cs
using MaksIT.Core.Abstractions.Webapi;


namespace MaksIT.CertsUI.Models.Identity.User;

public class CreateUserRequest : RequestModelBase {
  public required string Username { get; set; }
  public required string Email { get; set; }
  public required string MobileNumber { get; set; }
  public required string Password { get; set; }
  public bool IsGlobalAdmin { get; set; }
  public List<CreateUserEntityScopeRequest>? EntityScopes { get; set; }
}

File: src/MaksIT.WebUI/src/models/user/CreateUserRequest.ts
import { array, boolean, object, string, type ZodType } from 'zod'
import { RequestModelBase } from '@maks-it.com/webui-contracts'
import { withCreateGlobalAdminRefine } from '../engine/globalAdminZod'
import { CreateUserEntityScopeRequest, CreateUserEntityScopeRequestSchema } from './CreateUserEntityScopeRequest'

export interface CreateUserRequest extends RequestModelBase {
  username: string
  email: string
  mobileNumber: string
  password: string
  isGlobalAdmin?: boolean
  entityScopes?: CreateUserEntityScopeRequest[] | null
}

export const CreateUserRequestSchema: ZodType<CreateUserRequest> = object({
  username: string().min(1),
  email: string().email(),
  mobileNumber: string().min(1),
  password: string().min(8),
  isGlobalAdmin: boolean().optional(),
  entityScopes: array(CreateUserEntityScopeRequestSchema).optional().nullable(),
})

export const createCreateUserRequestSchema = (actorIsGlobalAdmin: boolean) =>
  withCreateGlobalAdminRefine(CreateUserRequestSchema, actorIsGlobalAdmin)
